easylist.go

```diff
@@ -20,6 +21,14 @@ const (
 
 var (
 	log = golog.LoggerFor("easylist")
+
+	// Find rules begin with "||" and end with "^", extract the string in
+	// between, except those followed by "$domain=", e.g.,
+	// "||s3.amazonaws.com^$domain=imgclick.net".
+	//
+	// Check https://adblockplus.org/filters for details.
+	re      = regexp.MustCompile(`^\|\|([\w.-]+\.\w+)\^`)
+	exclude = regexp.MustCompile(`domain=`)
```

Comment:
Rather than trying to parse the rules ourselves, maybe it makes more sense to just use [ParseRules](https://github.com/pmezard/adblock/blob/master/adblock/rules.go#L288) from the AdBlock library to actually do the parsing, and then extract the information from there that we need?